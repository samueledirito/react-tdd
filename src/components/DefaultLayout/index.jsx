import React from "react";
import { Route } from "react-router-dom";
import AppBar from "./AppBar";
import CssBaseline from "@material-ui/core/CssBaseline";

const DefaultLayout = ({ component: Component, ...rest }) => {
  return (
    <Route
      {...rest}
      render={(matchProps) => (
        <div className="DefaultLayout">
          <CssBaseline />
          <AppBar />
          <Component {...matchProps} />
        </div>
      )}
    />
  );
};

export default DefaultLayout;
