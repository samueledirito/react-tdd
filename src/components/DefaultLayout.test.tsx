import React from "react";
import { render, fireEvent } from "@testing-library/react";

import DefaultLayout from "components/DefaultLayout";

const TestComponent = () => <div>Test Component</div>;

describe("<DefaultLayout />", () => {
  let expectedSnapshot;
  let header;
  let content;
  let hamburger;
  let chevron;

  beforeEach(() => {
    const { getByTestId, asFragment } = render(
      <DefaultLayout>
        <TestComponent />
      </DefaultLayout>
    );
    header = getByTestId("header");
    content = getByTestId("content");
    hamburger = getByTestId("hamburger");
    chevron = getByTestId("chevron");
    expectedSnapshot = asFragment();
  });

  test("should be created", () => {
    expect(expectedSnapshot).toMatchSnapshot();
  });

  test("should contain header", () => {
    expect(header).toBeInTheDocument();
  });

  test("should contain chosen component", () => {
    expect(content).toBeInTheDocument();
  });

  test("drawer appears and disappears when clicking menu button", async () => {
    expect(hamburger).toBeVisible();
    fireEvent.click(hamburger);
    expect(hamburger).not.toBeVisible();

    fireEvent.click(chevron);
    expect(hamburger).toBeVisible();
  });
});
