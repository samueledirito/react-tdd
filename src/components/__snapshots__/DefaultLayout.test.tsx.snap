// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultLayout /> should be created 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        data-testid="header"
      >
        <div
          class="makeStyles-menuButton-4"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            data-testid="hamburger"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
          data-testid="app-title"
        >
          React Best Practices
        </h6>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-6"
      data-testid="drawer"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-7 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
      >
        <div
          class="makeStyles-drawerHeader-8"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="chevron"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-dense MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
              >
                Inbox
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
              >
                Starred
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
              >
                Send email
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
              >
                Drafts
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
        <hr
          class="MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-dense MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
              >
                All mail
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
              >
                Trash
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
              >
                Spam
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
    </div>
    <div
      class="MuiContainer-root makeStyles-content-9 MuiContainer-maxWidthLg"
      data-testid="content"
    >
      <div
        class="makeStyles-drawerHeader-8"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-8 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
        >
          <div>
            Test Component
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
